Editor/MenuActions/Geometry/DuplicateFaces.cs

```diff
@@ -0,0 +1,167 @@
+        public override TooltipContent tooltip { get { return s_Tooltip; } }
+
+        static readonly TooltipContent s_Tooltip = new TooltipContent
+            (
+                "Duplicate Faces",
+                "Creates a new object (or submesh) from the selected faces."
+            );
+
+        public override SelectMode validSelectModes
+        {
+            get { return SelectMode.Face; }
+        }
+
+        public override bool enabled
+        {
+            get { return base.enabled && MeshSelection.selectedFaceCount > 0; }
+        }
+
+        protected override MenuActionState optionsMenuState
+        {
+            get { return MenuActionState.VisibleAndEnabled; }
+        }
+
+        internal enum DuplicateFaceSetting
+        {
+            GameObject,
+            Submesh
+        };
+
+        protected override void OnSettingsGUI()
+        {
+            GUILayout.Label("Duplicate Face Settings", EditorStyles.boldLabel);
+
+            EditorGUILayout.HelpBox("Duplicate Faces can separate the selection into either a new GameObject or a submesh.", MessageType.Info);
+
+            EditorGUI.BeginChangeCheck();
+
+            m_DuplicateFaceSetting.value = (DuplicateFaceSetting)EditorGUILayout.EnumPopup("Duplicate To", m_DuplicateFaceSetting);
+
+            if (EditorGUI.EndChangeCheck())
+                ProBuilderSettings.Save();
+
+            GUILayout.FlexibleSpace();
+
+            if (GUILayout.Button("Duplicate Selection"))
+                EditorUtility.ShowNotification(DoAction().notification);
+        }
+
+        public override ActionResult DoAction()
+        {
+            if (MeshSelection.selectedObjectCount < 1)
+                return ActionResult.NoSelection;
+
+            UndoUtility.RecordSelection("Duplicate Face(s)");
+
+            if (m_DuplicateFaceSetting == DuplicateFaceSetting.GameObject)
+                return DuplicateFacesToObject();
+
+            return DuplicateFacesToSubmesh();
+        }
+
+        static ActionResult DuplicateFacesToSubmesh()
+        {
+            int count = 0;
+
+            foreach (ProBuilderMesh pb in MeshSelection.topInternal)
+            {
+                pb.ToMesh();
+                List<Face> res = pb.DetachFaces(pb.selectedFacesInternal, false);
+                pb.Refresh();
+                pb.Optimize();
+
+                pb.SetSelectedFaces(res.ToArray());
+
+                count += pb.selectedFaceCount;
+            }
+
+            ProBuilderEditor.Refresh();
+
+            if (count > 0)
+                return new ActionResult(ActionResult.Status.Success, "Duplicate " + count + (count > 1 ? " Faces" : " Face"));
+
+            return new ActionResult(ActionResult.Status.Success, "Duplicate Faces");
+        }
+
+        static ActionResult DuplicateFacesToObject()
+        {
+            int DuplicateedFaceCount = 0;
+            List<GameObject> Duplicateed = new List<GameObject>();
```

Comment:
extra `e` in Duplicated in these variables

Reply:
Done thanks!